mackerel-plugin-postgres/postgres.go

```diff
@@ -180,15 +189,54 @@ func fetchDatabaseSize(db *sql.DB) (map[string]float64, error) {
 	return stat, nil
 }
 
-func mergeStat(dst, src map[string]float64) {
+func (p PostgresPlugin) fetchRelpages(db *sql.DB) (map[string]float64, error) {
+	if p.RelpagesLimit <= 0 {
+		return nil, nil
+	}
+	if p.RelpagesAnalyze {
+		_, err := db.Query("analyze")
+		if err != nil {
+			logger.Warningf("Failed to ANALYZE", err)
+			return nil, err
+		}
+	}
+	rows, err := db.Query("select relname, relpages from pg_class order by relpages desc limit $1", p.RelpagesLimit)
+	if err != nil {
+		logger.Errorf("Failed to SELECT pg_class.relpages. %s", err)
+		return nil, err
+	}
+
+	stat := make(map[string]float64)
+
+	for rows.Next() {
+		var relname string
+		var relpages int64
+		if err := rows.Scan(&relname, &relpages); err != nil {
+			logger.Warningf("Failed to scan %s", err)
+			continue
+		}
+		stat["postgres.relpages."+p.Database+"."+relname] = float64(relpages)
```

Comment:
- According to META, key prefix `postgres.` is unnecessary
- All metrics provided by this plugin is about `p.Database`, so it seems meaningless.